substratevm/src/com.oracle.graal.pointsto/src/com/oracle/graal/pointsto/heap/HeapSnapshotVerifier.java

```diff
@@ -106,14 +106,21 @@ public boolean requireAnalysisIteration(CompletionExecutor executor) throws Inte
         return analysisModified || verificationReachableTypes > 0;
     }
 
+    protected ObjectScanner installObjectScanner(CompletionExecutor executor) {
+        return new ObjectScanner(bb, executor, scannedObjects, new ScanningObserver());
+    }
+
     protected void scanTypes(@SuppressWarnings("unused") ObjectScanner objectScanner) {
     }
 
     public void cleanupAfterAnalysis() {
         scannedObjects = null;
     }
 
-    private final class ScanningObserver implements ObjectScanningObserver {
+    protected final class ScanningObserver implements ObjectScanningObserver {
+
+        public ScanningObserver() {
+        }
```

Comment:
Why do you need to declare the default constructor?

Reply:
Because it is accessed by the `StandaloneHeapSnapshotVerifier`(https://github.com/oracle/graal/pull/6696/files#diff-42a787ac539b2d41b4211ecf9b92b29051d4a79e24ccb1257e43b2a28c3a83b2R45).